### PR TITLE
feature: support websocket control channel

### PR DESCRIPTION
**Pull request by piscisaureus**
connected to strongloop-internal/scrum-nodeops#449

### CODE REVIEW

**Comment by piscisaureus** [comment]
Please review

File: lib/runctl.js [hunk 1]
```diff
@@ -27,7 +27,12 @@ var server = require('strong-control-channel/server').create(onRequest);
 // attach targetctl to the master).
 var ipcctl = null;
 if (cluster.isMaster && process.send) {
```
Comment (rmg) [inline lib/runctl.js]: Thanks to the code below, this case no longer depends on `process.send`.

**Comment by rmg** [comment]
Could definitely use at least a minimal test, probably verifying the `started` notification is sent to the ws server it connects to.

File: lib/runctl.js (hunk 1 above)
Comment (sam-github) [inline lib/runctl.js]: And if supervisor isn't a direct child process, it won't have process.send: docker images, for example.

**Comment by sam-github** [comment]
@rmg added unit tests, they depend on strongloop/strong-control-channel#19

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by rmg** [comment]
@sam-github looks like the tests are timing out.

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Pull request merged**